core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -143,7 +143,9 @@ public User addUserCreatedOnBuddyRequest(UserDTO buddyUser, String tempPassword)
 	{
 		UUID savedUserID = CryptoSession.execute(Optional.of(tempPassword), null,
 				() -> tempEncryptionContextExecutor.addUserCreatedOnBuddyRequest(buddyUser).getID());
-		return getEntityByID(savedUserID);
+		User userEntity = getEntityByID(savedUserID);
+		sendMobileNumberConfirmationMessage(userEntity);
```

Comment:
The SMS should be sent after the saved the data under the device password.
In that stage, the app should have informed the user.